pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -970,7 +970,8 @@ func validateEmulatedMachine(field *k8sfield.Path, spec *v1.VirtualMachineInstan
 		supportedMachines := config.GetEmulatedMachines(spec.Architecture)
 		var match = false
 		for _, val := range supportedMachines {
-			if regexp.MustCompile(val).MatchString(machine.Type) {
+			// The pattern are hardcoded, so this should not throw an error
+			if ok, _ := filepath.Match(val, machine.Type); ok {
```

Comment:
From the usage of `filepath` it looks like we do care about the formatting of the machine type field?
But OTOH we ignore the error, why wouldn't we use the regular comparison `val == machine.Type` then?

Reply:
Because the value is a glob pattern for x86 it would be `q35*`. Because as machine type it could be something like `q35-rhel-9.1` or so (i wrote it from memory, so it might be spelled differently).
So a simple `==` does not work, while the previous regex just ensured that `q35` is somewhere in the string. We could hardcode possible values, but i don't know where we would find them and we would need to constantly update it.

The full list of patterns from `pkg/virt-config/virt-config.go`:
```
	DefaultAMD64EmulatedMachines                    = "q35*,pc-q35*"
	DefaultPPC64LEEmulatedMachines                  = "pseries*"
	DefaultAARCH64EmulatedMachines                  = "virt*"
	DefaultS390XEmulatedMachines                    = "s390-ccw-virtio*"
```